cmd/client/main.go

```diff
@@ -241,17 +242,43 @@ func (c *Client) run(o *GrpcProxyClientOptions) error {
 	// So to prevent leaking too many tunnel connections,
 	// we explicitly close the tunnel on the first CLOSE_RSP we obtain from the inner TCP connection
 	// (https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/konnectivity-client/pkg/client/client.go#L137).
+	var wg sync.WaitGroup
+
 	for i := 1; i <= o.testRequests; i++ {
-		err = c.makeRequest(o, client)
-		if err != nil {
-			return err
-		}
+		wg.Add(1)
+		go func() error {
+			defer wg.Done()
+			dialer, err := c.getDialer(o)
+			if err != nil {
+				return fmt.Errorf("failed to get dialer for client, got %v", err)
+			}
+			transport := &http.Transport{
+				DialContext: dialer,
+			}
+			client := &http.Client{
+				Transport: transport,
+			}
```

Comment:
This changes the behavior of the client from reusing the same underlying connection to creating a new connection for each http request. I think both are ok. Do you change the behavior on purpose?

Reply:
It was. As per the comment header "The apiserver network proxy does not support reusing the same dialer for multiple connections." So was attempting to ensure the test client reflects our primary use case. Up to this point our load was serial so it mattered less but as I was making the requests concurrent it was better to stay consistent, If we want a reuse dialer mode, I'm fine with adding it. I think that needs to be given some thought for the model to be correct. Also we should probably understand the requirement for that use case first.